contracts/AttestationRegistry.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.12;
+
+contract AttestationRegistry {
+    struct Attestation {
+        bytes32 id; // Holds up to a 256-bit hash
+        bytes32 revision; // Holds up to a 256-bit hash
+        bytes32 publicKey; // Holds an ed25519 key
+        bytes1[64] signature; // Holds an ed25519 signature
+    }
+
+    /// The set of recorded attestations. Maps between an Ethereum address
+    /// and an attestation.
+    mapping(address => Attestation) public attestations;
+
+    /// Create a new attestation. Overwrites any existing attestation by
+    /// the sender.
+    function attest(
+        bytes32 id,
+        bytes32 revision,
+        bytes32 publicKey,
+        bytes1[64] calldata signature
+    ) public {
+        attestations[msg.sender].id = id;
+        attestations[msg.sender].revision = revision;
+        attestations[msg.sender].publicKey = publicKey;
+        attestations[msg.sender].signature = signature;
+    }
+
+    /// Revoke any attestation made by the sender.
+    function revokeAttestation() public {
```

Comment:
One thing to meditate over is if we'd want to provide a way to issue a _signed_ revocation. That is, if I still have access to a valid `link` key, migrating to a new identity would be verifiably legit. Doing so may or may not make sense, though

Reply:
Ah that's a good point. I think both options will probably be needed, in case you lose your *link* key. Clients would have to handle them differently though :thinking: